logger/options.go

```diff
@@ -40,43 +39,62 @@ type Options struct {
 	Timestamp     time.Time
 }
 
+type optionsError struct {
+	fields []string
+	probs  []string
+}
+
 func (e InvalidOptionMessage) String() string {
 	return fmt.Sprintf("Options.%s: %s", e.Option, e.Message)
 }
 
-func validateOptionLength(option string, value string, problems *[]string) {
-	if len(value) > 32 {
-		*problems = append(*problems, InvalidOptionMessage{option, "length must be less than 32"}.String())
+func validateOptionLength(option string, value string, myErr optionsError) {
+	if len(value) > 80 {
+		myErr.fields = append(myErr.fields, option)
+		myErr.fields = append(myErr.probs, "length must be less than 80")
 	}
 }
 
+func (e *optionsError) Error() string {
+	return fmt.Sprintf("%s: %s", e.fields, e.probs)
+}
+
 func (options *Options) validate() error {
-	var problems []string
-	reMacAddress := regexp.MustCompile(`^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])`)
 	reHostname := regexp.MustCompile(`(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])`)
-	reIPAddress := regexp.MustCompile(`(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}`)
-
-	validateOptionLength("App", options.App, &problems)
-	validateOptionLength("Env", options.Env, &problems)
-	validateOptionLength("Hostname", options.Hostname, &problems)
-	validateOptionLength("Level", options.Level, &problems)
 
-	if options.MacAddress != "" && (!reMacAddress.MatchString(options.MacAddress)) {
-		problems = append(problems, InvalidOptionMessage{"MacAddress", "invalid format"}.String())
+	myErr := optionsError{}
+	// myErr.fields = make([]string, 10)
+	// myErr.probs = make([]string, 10)
+
+	validateOptionLength("App", options.App, myErr)
+	validateOptionLength("Env", options.Env, myErr)
+	validateOptionLength("Hostname", options.Hostname, myErr)
+	validateOptionLength("Level", options.Level, myErr)
+
+	if options.MacAddress != "" {
+		mac, err := net.ParseMAC(options.MacAddress)
+		if err != nil {
+			myErr.fields = append(myErr.fields, "MacAddress")
+			myErr.probs = append(myErr.probs, "invalid format")
+		} else {
+			options.MacAddress = mac.String()
+		}
 	}
-
 	if options.Hostname != "" && !reHostname.MatchString(options.Hostname) {
-		problems = append(problems, InvalidOptionMessage{"Hostname", "invalid format"}.String())
+		myErr.fields = append(myErr.fields, "Hostname")
+		myErr.probs = append(myErr.probs, "invalid format")
 	}
-
-	if options.IPAddress != "" && !reIPAddress.MatchString(options.IPAddress) {
-		problems = append(problems, InvalidOptionMessage{"IPAddress", "invalid format"}.String())
+	if options.IPAddress != "" && net.ParseIP(options.IPAddress) == nil {
+		myErr.fields = append(myErr.fields, "IPAddress")
+		myErr.probs = append(myErr.probs, "invalid format")
+		fmt.Println(myErr)
 	}
 
-	if len(problems) > 0 {
-		return errors.New(strings.Join(problems, ", "))
-	}
+	fmt.Println(myErr)
```

Comment:
@jakedipity @mdeltito am attempting @jakedipity's suggestion with errors- and it looks like something is going wrong because myErr is consistently an empty struct at this point. I tried making the slices above but getting the same result- also this is causing it to error out with CI